app-api/src/main/resources/liquibase-sender.xml

```diff
@@ -250,5 +250,74 @@
             </column>
         </addColumn>
     </changeSet>
+    
+    <changeSet author="lpinto" id="20220210-1050">
+        <preConditions onFail="MARK_RAN">
+            <not>
+                <tableExists tableName="sender_sync_message" />
+            </not>
+        </preConditions>
+        <comment>Adding sender_sync_message table</comment>
+
+        <createTable tableName="sender_sync_message">
+            <column autoIncrement="true" name="id" type="BIGINT">
+                <constraints nullable="false" primaryKey="true" />
+            </column>
+            <column name="table_name" type="VARCHAR(100)">
+                <constraints nullable="false" />
+            </column>
+            <column name="identifier" type="VARCHAR(255)">
+                <constraints nullable="false" />
+            </column>
+            <column name="operation" type="VARCHAR(1)">
+                <constraints nullable="false"/>
+            </column>
+            <column name="message_uuid" type="VARCHAR(38)">
+                <constraints nullable="false" unique="true"/>
+            </column>
+            <column name="request_uuid" type="VARCHAR(38)" />
+            <column name="date_created" type="DATETIME">
+                <constraints nullable="false" />
+            </column>
+            <column name="snapshot" type="BOOLEAN">
+                <constraints nullable="false" />
+            </column>
+            <column name="status" type="VARCHAR(20)">
+                <constraints nullable="false" />
+            </column>
+            <column name="date_updated" type="DATETIME" />
+            <column name="error_message" type="VARCHAR(200)" />
```

Comment:
Please remove this error_message column

Reply:
There was a purpose behind the creation of this column. I think that the better approach is to try to understand it before asking for removal. If this is not the correct way of doing things and after the explanation the decision to remove remains, then we will remove.

Please see [this comment](https://github.com/FriendsInGlobalHealth/openmrs-eip/pull/11#discussion_r907182671)

Reply:
It is not an error if the entity is not found, the column is not required and currently the original logic is broken when I tested because the receiver is not getting the response back to resolve the sync request so it stays hanging forever, please don't change the original code, let's not introduced an new unnecessary functionalities or columns for as part of this work

Reply:
I added a comment to respond to you, the TODO is for something else

Reply:
Like I said before, I did not changed the logic for sync requests. When no entity is found we still send an empty response, as you can see here:
https://github.com/FriendsInGlobalHealth/openmrs-eip/blob/faa9961347844922604c8f13d3ebd77550560298/distribution/sender/routes/sender-activemq-publisher.xml#L66-L80

But my question remains:
When we receive a database event, we create a SenderSyncMessage with the status 'NEW'.
Then, sender-activemq-publisher processes all the SenderSyncMessage with the status 'NEW'.
So, when we have a SenderSyncMessage from a regular database change (not a sync request) for which we do not found the correspondent entity, what do we do? Do we change the status or delete it? Otherwise it will be reprocessed in the sender-activemq-publisher forever.

Reply:
If we don't find an entity for a database event, usually this is because the entity was deleted by another event and it is not an error, we just log the message, ignore that TODO, in fact remote it.

Reply:
Delete the sync message row if the entity is not found when processing a regular DB event